nautobot/core/tests/test_api.py

```diff
@@ -401,7 +401,9 @@ def test_related_by_pk(self):
 
         response = self.client.post(url, data, format="json", **self.header)
         self.assertHttpStatus(response, status.HTTP_201_CREATED)
-        self.assertEqual(str(response.data["location"]), str(self.location1.pk))
+        self.assertEqual(
+            str(response.data["location"]), f"http://nautobot.example.com{self.location1.get_absolute_url(api=True)}"
```

Comment:
Instead of hard-coding `http://nautobot.example.com{self.location1.get_absolute_url(api=True)}` in a bunch of places, it would be cleaner to use a request factory, and stash this in a helper method on the base test class.

e.g. 

```python
from django.test import RequestFactory

def get_api_url(self, obj):
    request = RequestFactory().request(self.client._base_environ()["SERVER_NAME"])
    return request.build_absolute_uri(obj.get_absolute_url(api=True))
```

And then have the test code call `self.get_api_url(obj)`